src/Codeception/Util/JsonArray.php

```diff
@@ -121,11 +121,33 @@ private function arrayToXml(DOMDocument $doc, DOMNode $node, array $array): void
             if (is_array($value)) {
                 $this->arrayToXml($doc, $subNode, $value);
             } else {
-                $subNode->nodeValue = htmlspecialchars((string)$value);
+                $this->setValue($subNode, $value);
             }
         }
     }
 
+    private function setValue($subNode, $value) {
+        switch(gettype($value)) {
+            case "boolean":
+                $subNode->nodeValue = $value?"true":"false";
```

Comment:
Please format this line as `$subNode->nodeValue = $value ? 'true' : 'false';`

and use single quotes in all strings in this method.

Reply:
ok